src/hooks/screen/useChatScreenHook.ts

```diff
@@ -123,6 +123,25 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
     }
   };
 
+  const handleChatData = async () => {
+    if (!localDb || !selectedChannel) return;
+    if (!otherListener[`chat_${selectedChannel?.id}`]) return;
+
+    try {
+      getAllMessages.refetch();
+      if (type === 'ANONYMOUS') {
+        const userInfo = await UserSchema.getSelfAnonUserInfo(
+          localDb,
+          anonProfileId,
+          selectedChannel?.id
+        );
+        setSelfAnonUserInfo(userInfo);
+      }
+    } catch (e) {
+      console.log(e, 'error get all chat');
+    }
+  };
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

A new function `handleChatData` is introduced to manage chat data. It fetches all messages and sets user info for anonymous chats. Error handling is done using a try-catch block which logs the error message. This function improves the modularity of the code by separating the logic for managing chat data.
```javascript
const handleChatData = async () => {
  if (!localDb || !selectedChannel) return;
  if (!otherListener[`chat_${selectedChannel?.id}`]) return;

  try {
    getAllMessages.refetch();
    if (type === 'ANONYMOUS') {
      const userInfo = await UserSchema.getSelfAnonUserInfo(
        localDb,
        anonProfileId,
        selectedChannel?.id
      );
      setSelfAnonUserInfo(userInfo);
    }
  } catch (e) {
    console.log(e, 'error get all chat');
  }
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->